src/main/java/teammates/logic/core/FeedbackResponseCommentsLogic.java

```diff
@@ -269,50 +267,51 @@ public boolean isResponseCommentVisibleForUser(String userEmail, String courseId
             isVisibleResponseComment = true;
         } else if (role == Role.STUDENT 
                 && ((relatedQuestion.recipientType == FeedbackParticipantType.TEAMS
-                        && isResponseCommentVisibleTo(relatedQuestion, relatedComment, FeedbackParticipantType.RECEIVER))
+                        && isResponseCommentVisibleTo(relatedQuestion, relatedComment,
+                                                      FeedbackParticipantType.RECEIVER))
                         && response.recipientEmail.equals(student.team))
                     || ((relatedQuestion.giverType == FeedbackParticipantType.TEAMS
-                        || isResponseCommentVisibleTo(relatedQuestion, relatedComment, FeedbackParticipantType.OWN_TEAM_MEMBERS))
+                        || isResponseCommentVisibleTo(relatedQuestion, relatedComment,
+                                                      FeedbackParticipantType.OWN_TEAM_MEMBERS))
                             && studentsEmailInTeam.contains(response.giverEmail))
-                    || (isResponseCommentVisibleTo(relatedQuestion, relatedComment, FeedbackParticipantType.RECEIVER_TEAM_MEMBERS)
+                    || (isResponseCommentVisibleTo(relatedQuestion, relatedComment,
+                                                   FeedbackParticipantType.RECEIVER_TEAM_MEMBERS)
                             && studentsEmailInTeam.contains(response.recipientEmail))) {
```

Comment:
See if you can extract each condition into an explanatory variable (i.e. apply 'introduce explaining variable' refactoring).